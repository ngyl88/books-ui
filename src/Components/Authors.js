import React, { Component } from "react";
import { API_HOST } from "../config";

class Authors extends Component {
  constructor() {
    super();
    this.state = {
      authors: []
    };
  }
  componentDidMount() {
      this.getAuthors();
  }
  render() {
    return (
      <div>
        <h3>Authors</h3>
        {this.state.authors.map(author => (
          <li key={author._id}>{author.name}</li>
        ))}
      </div>
    );
  }
  async getAuthors() {
    const response = await fetch(`${API_HOST}/authors`);
    const data = response.ok ? await response.json() : [];
    this.setState({
      authors: data
    });
  }
}

export default Authors;
