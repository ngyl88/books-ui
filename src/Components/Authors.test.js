import React from "react";
import ShallowRenderer from "react-test-renderer/shallow";
import Authors from "./Authors";

it("renders title", () => {
  const renderer = new ShallowRenderer();
  renderer.render(<Authors />);

  const output = renderer.getRenderOutput();
  expect(output).toMatchSnapshot();
});

it("renders title and list of authors after component mounts", async () => {
  fetch.mockResponseOnce(
    JSON.stringify([
      {
        _id: "20000001",
        name: "author 1"
      }
    ])
  );

  const renderer = new ShallowRenderer();
  renderer.render(<Authors />);

  const instance = renderer.getMountedInstance();
  await instance.componentDidMount();

  const output = renderer.getRenderOutput();
  expect(output).toMatchSnapshot();
});
