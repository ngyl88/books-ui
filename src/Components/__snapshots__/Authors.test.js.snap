// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders title 1`] = `
<div>
  <h3>
    Authors
  </h3>
</div>
`;

exports[`renders title and list of authors after component mounts 1`] = `
<div>
  <h3>
    Authors
  </h3>
  <li>
    author 1
  </li>
</div>
`;
